djactasauth/backends.py

```diff
@@ -60,19 +60,33 @@ def authenticate(self, username=None, password=None, **kwargs):
                 username=username, password=password, **kwargs)
 
     def _authenticate(self, username=None, password=None, **kwargs):
-        if self.is_act_as_username(username):
-            auth_username, act_as_username = username.split(self.sepchar)
-        else:
-            auth_username = act_as_username = username
-
-        auth_user = super(ActAsModelBackend, self)._authenticate(
-                username=auth_username, password=password, **kwargs)
-        if not auth_user:
-            return auth_user
-        if auth_username != act_as_username:
-            UserModel = get_user_model()
+        self.fail_unless_one_aaa_backend_is_configured()
+        assert password is not None
+        if not self.is_act_as_username(username):
+            return None
+        for backend in auth.get_backends():
+            if not isinstance(backend, ActAsBackend):
```

Comment:
* I feel the mix of short and long variable names on the same line isn't nice 
* I've seen a number of people get confused at first over nested list comprehensions in the past
* alternatively, to avoid the nesting, the if could be reversed, i.e.: `if isinstance(backend, ActAsBackend): continue`, but many people have strong opinions against early returns (though I don't)

in the light of the above (plus the fact that @kairichard has already merged it), I would leave it as is, and I have a feeling next time I'm in this area, I'll do a bit refactoring over the complex methods so I can lower `flake8`'s complexity limit even further down, and extracting this loop/if into its own method would be a nice way to address both of the concerns